ncm-dpmlfc/src/main/perl/DPMLFC/RuleBasedEditor.pm

```diff
@@ -0,0 +1,778 @@
+# ${license-info}
+# ${developer-info}
+# ${author-info}
+# ${build-info}
+#
+#
+# This module implement a rule-based editor that is used to modify the content
+# of an existing file without taking care of the whole file. Each rule
+# driving the edition process is applied to one matching line. The input for
+# updating the file is the Quattor configuration and conditions can be defined
+# based on the contents of this configuration.
+#
+#######################################################################
+
+package NCM::Component::DPMLFC::RuleBasedEditor;
+
+use strict;
+use NCM::Component;
+use vars qw(@ISA $EC);
+@ISA = qw(NCM::Component);
+$EC=LC::Exception::Context->new->will_store_all;
+
+use EDG::WP4::CCM::Element;
+
+use Readonly;
+
+use File::Path;
+use File::Copy;
+use File::Compare;
+use File::Basename;
+use File::stat;
+
+use LC::Check;
+use CAF::FileWriter;
+use CAF::FileEditor;
+use CAF::FileReader;
+use CAF::Process;
+use CAF::Object;
+
+use Encode qw(encode_utf8);
+use Fcntl qw(SEEK_SET);
+
+local(*DTA);
+
+
+# Constants use to format lines in configuration files
+# Keep in sync with ncm-xrootd
+use constant LINE_FORMAT_PARAM => 1;
```

Comment:
I don't know why but I prefer constants for this. With the approach used, it should be easy to manage and don't require any particular variable declaration in the code using it. Currently constants are not exported by default because the trick used with 'use parent' is causing a conflict but after moving it to CAF, I think those constants should be exported by default by `FileEditor` for example.